core/src/main/scala/cats/instances/ordering.scala

```diff
@@ -0,0 +1,16 @@
+package cats
+package instances
+
+import cats.functor.Contravariant
+
+trait OrderingInstances {
+
+  implicit val catsFunctorContravariantForOrdering: Contravariant[Ordering] =
+    new Contravariant[Ordering] {
+      /** Derive an `Ordering` for `B` given an `Ordering[A]` and a function `B => A`.
+       *
+       * Note: resulting instances are law-abiding only when the functions used are injective (represent a one-to-one mapping)
+       */
```

Comment:
On a related note (not suggesting we should change anything here), we do have an `Inject` with laws, user could use it to test if a function `f: A => B` is injective, by creating a `prj: B => Option[A]` and an `Inject[A, B]` using these two, then testing the law against this `Inject` instance. 
```scala
{
  implicit val inject = new Inject[A, B] { 
     val inj = f 
     val prj : B => Option[A] = ???
  }
 checkAll("Inject[A, B]", InjectTests[A, B].inject)
}
```
If you think this would be a good idea, we could potentially added to the `cats-testkit`.